internal/textwrap/textwrap_test.go

```diff
@@ -0,0 +1,213 @@
+package textwrap
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestDedent(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
+			name: "Basic indentation",
+			input: `
+\t\t\tHello
+\t\t\t\tWorld
+\t\t\t  Foo
+\t\t\t    Bar
+\t\t\t`,
+			expected: `
+Hello
+\tWorld
+  Foo
+    Bar
+`,
+		},
+		{
+			name: "No common indentation",
+			input: `
+Hello
+\tWorld
+  Foo
+    Bar
+`,
+			expected: `
+Hello
+\tWorld
+  Foo
+    Bar
+`,
+		},
+		{
+			name: "All lines indented",
+			input: `
+\t\t\tHello
+\t\t\tWorld
+\t\t\tFoo
+\t\t\tBar
+\t\t\t`,
+			expected: `
+Hello
+World
+Foo
+Bar
+`,
+		},
+		{
+			name: "Mixed tabs and spaces",
+			input: `
+\t\t\t\tHello
+\t\t\t  \tWorld
+\t\t\t\t  Foo
+\t\t\t\t \tBar
+\t\t\t`,
+			expected: `
+\tHello
+  \tWorld
+\t  Foo
+\t \tBar
+`,
+		},
+		{
+			name: "Empty lines",
+			input: `
+\t\t\tHello
+
+\t\t\tWorld
+
+\t\t\tFoo
+\t\t\tBar
+\t\t\t`,
+			expected: `
+Hello
+
+World
+
+Foo
+Bar
+`,
+		},
+		{
+			name: "Only whitespace lines",
+			input: `
+\t\t\t
+\t\t\t\t
+\t\t\t  
+\t\t\t`,
+			expected: `
+
+
+
+`,
+		},
+		{
+			name:     "Single line",
+			input:    "    Hello World",
+			expected: "Hello World",
+		},
+		{
+			name: "Dedent more than common prefix",
+			input: `
+\t\t\t\tHello
+\t\t\tWorld
+\t\t\t\tFoo
+\t\t\tBar
+\t\t\t`,
+			expected: `
+\tHello
+World
+\tFoo
+Bar
+`,
+		},
+		{
+			name:     "Empty string",
+			input:    "",
+			expected: "",
+		},
+		{
+			name:     "String with only newlines",
+			input:    "\n\n\n",
+			expected: "\n\n\n",
+		},
+		{
+			name:     "String with line feed",
+			input:    "\n\r\n",
+			expected: "\n\r\n",
+		},
+		{
+			name: "Indentation with non-space/tab characters",
+			input: `
+\u00a0Hello
+\u00a0~\tWorld
+\u00a0\tFoo
+\u00a0\tBar
+\u00a0`,
+			expected: `
+\u00a0Hello
+\u00a0~\tWorld
+\u00a0\tFoo
+\u00a0\tBar
+\u00a0`,
+		},
+		{
+			name: "Very long lines",
+			input: `
+\t\t\t\tThis is a very long line that goes on and on and on and on and on and on and on and on and on and on and on and on and on and on and on
+\t\t\t\t\tThis is another very long line that goes on and on and on and on and on and on and on and on and on and on and on and on and on and on
+\t\t\t\tThis is a third very long line that goes on and on and on and on and on and on and on and on and on and on and on and on and on and on and on
+\t\t\t\t`,
+			expected: `
+This is a very long line that goes on and on and on and on and on and on and on and on and on and on and on and on and on and on and on
+\tThis is another very long line that goes on and on and on and on and on and on and on and on and on and on and on and on and on and on
```

Comment:
what's up with this `\t`?

Reply:
Was a pain in the ass to figure out which were tabs and spaces in my editor but it's pretty important this test include combinations of. So I added an escape expander to the test harness.

Reply:
is there another way that can be explained to me :confused: ? just looking at this test, I wouldn't expect

`\t\t\t\t\tThis is another...` to become `\tThis is another`

Reply:
Ah it's a little hard to see with wrapped lines but this line has 5 tabs while the least common indent is 4 tabs.

Reply:
oooooooooooh. Gotcha. dedent, not ltrim or whatever.

Reply:
righhht yeah it's aimed at making in-code string literals line up better with the indentation of the surrounding code